gridstatus/ieso.py

```diff
@@ -0,0 +1,307 @@
+import io
+import time
+import xml.etree.ElementTree as ET
+
+import pandas as pd
+import requests
+
+from gridstatus import utils
+from gridstatus.base import ISOBase, NotSupported
+from gridstatus.decorators import support_date_range
+from gridstatus.gs_logging import log
+
+# Load goes back several decades, with each year in one file.
+EARLIEST_LOAD = pd.Timestamp("2002-05-01").date()
+LOAD_INDEX_URL = "http://reports.ieso.ca/public/DemandZonal/"
+
+# The yearly file is updated daily, and contains data from the start of the year
+# through the current date.
+LOAD_TEMPLATE_URL = f"{LOAD_INDEX_URL}/PUB_DemandZonal_YYYY.csv"
+
+
+LOAD_FORECAST_INDEX_URL = "http://reports.ieso.ca/public/OntarioZonalDemand/"
+
+# Each forecast file contains data from the day in the filename going forward for
+# 34 days. The most recent file does not have a date in the filename.
+LOAD_FORECAST_TEMPLATE_URL = (
+    f"{LOAD_FORECAST_INDEX_URL}/PUB_OntarioZonalDemand_YYYYMMDD.xml"
+)
+
+# The farthest in the past that forecast files are available
+MAXIMUM_DAYS_IN_PAST_FOR_LOAD_FORECAST = 90
+# The farthest in the future that forecasts are available. Note that there are not
+# files for these future forecasts, they are in the current day's file.
+MAXIMUM_DAYS_IN_FUTURE_FOR_LOAD_FORECAST = 34
+
+
+INTERVAL_DURATION = pd.Timedelta(hours=1)
+
+
+class IESO(ISOBase):
+    """Independent Electricity System Operator (IESO)"""
+
+    name = "Independent Electricity System Operator"
+    iso_id = "ieso"
+
+    # All data is provided in EST, and does not change with DST. This means there are
+    # no repeated or missing hours in the raw data and we can safely use tz_localize
+    # without setting ambiguous or nonexistent times.
+    # https://www.ieso.ca/-/media/Files/IESO/Document-Library/engage/ca/ca-Introduction-to-the-Capacity-Auction.ashx
+    default_timezone = "EST"
+
+    status_homepage = "https://www.ieso.ca/en/Power-Data"
```

Comment:
This isn't really a status page. Maybe we should hit the data directory (http://reports.ieso.ca/public/)?

Reply:
what you did works for now. not currently used anywhere on gridstatus.io. just a pointer for users of the library